src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -1,14 +1,177 @@
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.ParseRequestService;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.PersistenceService;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.ValidationModelService;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 public class DeleteRoleAssignmentOrchestrator {
-    //1. call parse request service
-    //2. Call persistence service to store the created records
-    //3. Call Validation model service to create aggregation objects and apply drools validation rule
-    //5. For Each: If success then call persistence service to update assignment record status
-    //6. once all the assignment records are approved call persistence to update request status
-    //7. Call persistence to move assignment records to Deleted status
-    //8. Call the persistence to remove assignment records from RoleAssignmentLive table
+
+
+    private PersistenceService persistenceService;
+    private ParseRequestService parseRequestService;
+    private ValidationModelService validationModelService;
+    RequestEntity requestEntity;
+    AssignmentRequest assignmentRequest;
+
+    public DeleteRoleAssignmentOrchestrator(PersistenceService persistenceService,
+                                            ParseRequestService parseRequestService,
+                                            ValidationModelService validationModelService) {
+        this.persistenceService = persistenceService;
+        this.parseRequestService = parseRequestService;
+        this.validationModelService = validationModelService;
+    }
+
+    public ResponseEntity<Object> deleteRoleAssignment(String actorId,
+                                                       String process,
+                                                       String reference) throws Exception {
+        List<RequestedRole> requestedRoles = null;
+
+        //1. create the request Object
+        Request request = parseRequestService.prepareRequestObject(process, reference);
+        assignmentRequest = new AssignmentRequest();
+
+        //2. Call persistence service to store only the request
+        requestEntity = persistenceService.persistRequest(request);
+        requestEntity.setHistoryEntities(new HashSet<>());
+        request.setId(requestEntity.getId());
+
+        //3. retrieve all assignment records based on actorId/process+reference
+        if (actorId != null) {
+            requestedRoles = persistenceService.getRoleAssignment(UUID.fromString(actorId));
+
+        } else if (process != null && reference != null) {
+            requestedRoles = persistenceService.getExistingRoleByProcessAndReference(
+                process,
+                reference,
+                Status.LIVE.toString()
+            );
+
```

Comment:
There should be an else block to throw 400 bad request saying actorId or process+reference must be provided.

Reply:
Done.